src/Middleware/WebSockets/src/Microsoft.AspNetCore.WebSockets.csproj

```diff
@@ -13,6 +13,7 @@
 
   <ItemGroup>
     <Reference Include="Microsoft.AspNetCore.Http.Extensions" />
+    <Reference Include="Microsoft.AspNetCore.Server.Kestrel.Core" />
```

Comment:
😢

Reply:
I think this is necessary to disable the request body rate limits. Since both of these assemblies are in the framework I don't think this causes any problems.

Reply:
I know why it's there, but it means we don't have the correct abstractions for disabling limits in a non-server specific way.

Reply:
Yes and no. No such feature is possible on the other servers so we never bothered abstracting it.

Reply:
Lets take the opportunity to do that now that we have a reason to.

Reply:
Do we really have a reason to considering no other servers support this? Once there's at least one other server that does, I can see adding the abstraction.

Reply:
Yes we do. I’m not with the dirtying the layering, approach. We don’t need to make an abstraction, but we need to move the feature interfaces out if kestrel. 

We should also not lose sight of our original design to build proper abstractions. HTTP.sys might end up supporting this.

Reply:
It'd be more compelling if we still shipped Microsoft.AspNetCore.WebSockets or even Microsoft.AspNetCore.Http.Extensions as a package, but I supposed it does keep our project files a little cleaner.

Reply:
Actually, why can't we implement `IHttpUpgradeFeature` for HTTP/2 and have the server automatically disable rate limits?

Reply:
Okay, maybe not `IHttpUpgradeFeature` since that would be misleading but an entirely new feature that's similar. Maybe `IHttpConnectWithProtocolFeature` or something.

Reply:
This is a discussion I was having with @Tratcher as well

Reply:
> Yes we do. I’m not with the dirtying the layering, approach. We don’t need to make an abstraction, but we need to move the feature interfaces out if kestrel.

Why move them now if the namespace is still `Microsoft.AspNetCore.Server.Kestrel.Core.Features;`?

If another server is someday able to implement this, we can find the right abstraction then. I doubt the API would end up the same.

Reply:
Until we merge the dlls into a single dll we should keep the layering. WebSockets shouldn't depend on Kestrel.

Reply:
I think using `IHttpMinRequestBodyDataRateFeature` instead of something first-class like the `IHttpUpgradeFeature` used for HTTP/1.1 WebSockets is poor layering anyway. Other servers may support HTTP/2 WebSockets without supporting per-request rate limit control.

Reply:
Perfect, even more reason to push this closer to the server.

Reply:
Ok, I've introduced a new IHttpConnectFeature for this scenario. How's it look?

Reply:
Much better, can ew delete this reference now?